python/ml4ir/applications/ranking/tests/test_create_dataset.py

```diff
@@ -0,0 +1,74 @@
+# Run from ml4ir/python directory as: python3 -m pytest ml4ir/applications/ranking/tests/test_create_dataset.py
+
+import unittest
+import os
+import random
+from argparse import Namespace
+import warnings
+warnings.filterwarnings("ignore")
+
+from ml4ir.base.io import file_io
+from ml4ir.base.io.logging_utils import setup_logging
+from ml4ir.applications.ranking.data.scripts.create_dataset import run_dataset_creation
+
+
+ROOT_DATA_DIR = "ml4ir/applications/ranking/tests/data/csv/train"
+FEATURE_CONFIG = "ml4ir/applications/ranking/tests/data/csv/synthetic/feature_config.yaml"
+OUTPUT_DIR = "ml4ir/applications/ranking/tests/test_output/synthetic"
+LOG_DIR = 'ml4ir/applications/ranking/tests/test_output_log/'
+
+
+class RankingCreateDatasetTest(unittest.TestCase):
+    def setUp(
+            self,
+            root_data_dir: str = ROOT_DATA_DIR,
+            feature_config: str = FEATURE_CONFIG,
+            output_dir: str = OUTPUT_DIR,
+            log_dir: str = LOG_DIR
+    ):
+        self.root_data_dir = root_data_dir
+        self.feature_config = feature_config
+        self.output_dir = output_dir
+        self.log_dir = log_dir
+
+        # Set up logging
+        file_io.make_directory(self.log_dir, clear_dir=True)
+        outfile: str = os.path.join(self.log_dir, "output_log.csv")
+        self.logger = setup_logging(reset=True, file_name=outfile, log_to_file=True)
+
+    def test_synthetic_data(self):
```

Comment:
@calliebradley I like the test. Concise and puts us in a good place to build on top of.